pkg/transport/kafka-client/kafka_producer_client.go

```diff
@@ -0,0 +1,118 @@
+package kafkaclient
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/go-logr/logr"
+	kclient "github.com/open-cluster-management/hub-of-hubs-kafka-transport/kafka-client/kafka-producer"
+	"github.com/open-cluster-management/hub-of-hubs-spec-transport-bridge/pkg/transport"
+)
+
+// NewHOHProducer returns a new instance of HOHProducer object.
+func NewHOHProducer(log logr.Logger) (*HOHProducer, error) {
+	kp := &HOHProducer{
+		deliveryChan:  make(chan kafka.Event),
+		stopChan:      make(chan struct{}, 1),
+		kafkaProducer: nil,
+		log:           log,
+	}
+
+	kafkaProducer, err := kclient.NewKafkaProducer(kp.deliveryChan)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kafka producer: %w", err)
+	}
+
+	kp.kafkaProducer = kafkaProducer
+
+	return kp, nil
+}
+
+// HOHProducer abstracts hub-of-hubs-kafka-transport kafka-producer's generic usage.
+type HOHProducer struct {
+	log           logr.Logger
+	kafkaProducer *kclient.KafkaProducer
+	deliveryChan  chan kafka.Event
+	stopChan      chan struct{}
+	startOnce     sync.Once
+	stopOnce      sync.Once
+}
+
+// deliveryHandler handles results of sent messages.
+// For now failed messages are only logged.
+func (p *HOHProducer) deliveryHandler(e *kafka.Event) {
+	switch ev := (*e).(type) {
```

Comment:
variable names here are not readable and therefore this function is very hard to follow even though it's only 10 lines.

we need to better name the variables. instead of `e` use kafkaEvent.
instead of `ev` use kafkaEventType or something like that.